File: python/024-bob/bob.py
def is_question(hey_bob):
    return hey_bob.strip()[-1] == '?'

def is_yelling(hey_bob):
    return hey_bob.upper() == hey_bob and any(s.isalpha() for s in hey_bob)

def is_silent(hey_bob):
    return hey_bob.strip() == ''

def response(hey_bob):
    if is_silent(hey_bob):
        return "Fine. Be that way!"
    elif is_question(hey_bob) and is_yelling(hey_bob):
        return "Calm down, I know what I'm doing!"
    elif is_question(hey_bob):
        return "Sure."
    elif is_yelling(hey_bob):
        return "Whoa, chill out!"
    else:
        return "Whatever."

File: typescript/013-bob/bob.ts
function isAQuestion(message: string): boolean {
  const cleanedMessage = message.trim();
  return cleanedMessage.charAt(cleanedMessage.length - 1) === '?';
}

function isAYell(message: string): boolean {
  return (message.toUpperCase() === message) &&
  (/[a-zA-Z]+/.test(message));
}

function isSilence(message: string): boolean {
  return message.trim() === '';
}

export function hey(message: string): string {
  const question = isAQuestion(message);
  const yell = isAYell(message);
  if (question && yell) {
    return 'Calm down, I know what I\'m doing!';
  } else if (yell) {
    return 'Whoa, chill out!';
  } else if (question) {
    return 'Sure.';
  } else if (isSilence(message)) {
    return 'Fine. Be that way!'
  } else {
    return 'Whatever.';
  }
}
